src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -388,9 +417,7 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
             if (!r) {
                 r = {};
             }
-            if (!r.hasOwnProperty(this.state.indexColumn)) {
-                r[this.state.indexColumn] = this.state.fetchedRowCount + idx;
-            }
+            r[this.state.indexColumn] = this.state.fetchedRowCount + idx;
```

Comment:
Generate index column numbers in the UI. This is necessary with the vscodeDataFrame.py change, since those slices are now indexed relative to themselves and not to the original variable. Tested with AML, this does not mess up their actual index columns. This also doesn't impact data frames since the indices all line up with what would've been generated in the backend.

Reply:
Thanks this answers my other question.